### PR TITLE
0.1/607 setup laravel data models

### PR DESCRIPTION
**Pull request by charlesstrange2**
Models, migrations and relationships.

### CODE REVIEW

**Comment by katjam** [comment]
As a general rule - I don't add any relationship methods until I need to use them That saves on unused code and unnecessary test writing.

File: database/migrations/2017_12_04_113055_create_children_table.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('children', function (Blueprint $table) {
+            $table->increments('id');
+            $table->dateTime('dob');
+            $table->boolean('born')->default(false);
```
Comment (katjam) [inline database/migrations/2017_12_04_113055_create_children_table.php]: Could this be appended? If there is a bith date in the past then yes! If not, no! ?

File: database/migrations/2017_12_04_113032_create_families_table.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('families', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('rvid')->nullable();
```
Comment (katjam) [inline database/migrations/2017_12_04_113032_create_families_table.php]: Maybe a comment // Rose Vouchers Id

File: database/migrations/2017_12_04_113404_create_notes_table.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('notes', function (Blueprint $table) {
+            $table->increments('id');
+            $table->text('content'); // 64k
+            $table->integer('family_id')->unsigned(); // FK Families
+            $table->integer('user_id')->unsigned(); // FK Users
```
Comment (katjam) [inline database/migrations/2017_12_04_113404_create_notes_table.php:20]: Is this the creator of the note?

**Review by katjam — APPROVED** [review]
Probably OK - but I'd think hard about not having the relationship methods until you need them... maybe comment out for now and when you finish this round delete the ones you did not need.

File: database/migrations/2017_12_04_113032_create_families_table.php (hunk 2 above)
Comment (charlesstrange2) [reply database/migrations/2017_12_04_113032_create_families_table.php]: Agreed. is not obvious.

File: database/migrations/2017_12_04_113055_create_children_table.php (hunk 1 above)
Comment (charlesstrange2) [reply database/migrations/2017_12_04_113055_create_children_table.php]: It's a flag to say "this person has been acknowledged as actually being 'born'". 
While the future DoB was considered. They wanted a flag to say "we've been told this person has been born". There may be a UI/Print notice to check with the family that.

File: database/migrations/2017_12_04_113404_create_notes_table.php (hunk 3 above)
Comment (charlesstrange2) [reply database/migrations/2017_12_04_113404_create_notes_table.php:20]: Yes. Associates a user with the note.

**Pull request merged**